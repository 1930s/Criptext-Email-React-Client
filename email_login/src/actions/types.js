export const User = {
  ADD_BATCH: 'ADD_USERS',
  ADD: 'ADD_USER'
};
