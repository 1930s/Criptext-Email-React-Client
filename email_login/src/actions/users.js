import { User } from './types';
import {
  closeLogin,
  createAccount,
  createContact,
  openCreateKeys,
  openMailbox
} from '../utils/electronInterface';
import { API_URL } from './../utils/const';
import ClientAPI from '@criptext/email-http-client';
const client = new ClientAPI(API_URL);

export const addUsers = users => {
  return {
    type: User.ADD_BATCH,
    users: users
  };
};

export const addUser = user => {
  return async () => {
    try {
      const serverResponse = await client.postUser(user);
      const responseStatus = serverResponse.status;
      if (responseStatus === 200) {
        const localResponse = await createLocalData(user);
        const userId = localResponse[0];
        const terminated = await createLocalSession(userId, user);
        if (terminated) {
          openCreateKeys(user);
          closeLogin();
        }
      }
    } catch (e) {
      //TO DO
    }
  };
};

const createLocalData = user => {
  const localData = {
    username: user.username,
    name: user.name
  };
  return createContact(localData);
};

const createLocalSession = (id, user) => {
  const sessionData = {
    sessionId: id,
    username: user.username
  };
  return createAccount(sessionData);
};

export const loginUser = user => {
  return async () => {
    try {
      const userCredentials = {
        username: user.username,
        password: user.password,
        deviceId: 1
      };
      const serverResponse = await client.login(userCredentials);
      const responseStatus = serverResponse.status;
      if (responseStatus === 200) {
        openMailbox();
        closeLogin();
      } else {
        alert(serverResponse.text);
      }
    } catch (e) {
      // To do
    }
  };
};
