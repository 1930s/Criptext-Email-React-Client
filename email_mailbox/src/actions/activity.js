import { Activity } from './types';

export const startLoadSync = () => {
  return {
    type: Activity.START_LOAD_SYNC
  };
};

export const stopLoadSync = () => {
  return {
    type: Activity.STOP_LOAD_SYNC
  };
};

export const startLoadThread = () => {
  return {
    type: Activity.START_LOAD_THREAD
  };
};

export const stopLoadThread = () => {
  return {
    type: Activity.STOP_LOAD_THREAD
  };
};
