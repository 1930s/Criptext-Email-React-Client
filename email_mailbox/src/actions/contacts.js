import { Contact } from './types';
import * as db from '../utils/electronInterface';

export const addContacts = contacts => {
  return {
    type: Contact.ADD_BATCH,
    contacts: contacts
  };
};

export const loadContacts = ids => {
  return async dispatch => {
    try {
      const response = await db.getContactByIds(ids);
      const contacts = {};
      response.forEach(element => {
        contacts[element.id] = element;
      });
      dispatch(addContacts(contacts));
    } catch (e) {
      // TO DO
    }
  };
};
