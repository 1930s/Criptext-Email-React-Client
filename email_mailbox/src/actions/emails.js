import { Email } from './types';
import {
  getEmailsByThreadId,
  setMuteEmailById
} from '../utils/electronInterface';

export const addEmails = emails => {
  return {
    type: Email.ADD_BATCH,
    emails: emails
  };
};

export const muteNotifications = emailId => {
  return {
    type: Email.MUTE,
    targetEmail: emailId
  };
};

export const loadEmails = threadId => {
  return async dispatch => {
    try {
      const response = await getEmailsByThreadId(threadId);
      const emails = {};
      response.forEach(element => {
        emails[element.id] = element;
      });
      dispatch(addEmails(emails));
    } catch (e) {
      // TO DO
    }
  };
};

export const muteEmailById = feed => {
  return async dispatch => {
    try {
      const valueToSet = feed.get('isMuted') === 1 ? false : true;
      await setMuteEmailById(feed.get('emailId'), valueToSet);
      dispatch(muteNotifications(feed.get('emailId')));
    } catch (e) {
      // To do
    }
  };
};
