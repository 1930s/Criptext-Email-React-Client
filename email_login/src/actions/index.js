import { addUser, addUsers } from './users';

export { addUser, addUsers };
