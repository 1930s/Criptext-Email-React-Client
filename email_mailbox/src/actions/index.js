import { addUsers, addUser } from './users';
import {
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  filterThreadsByUnread,
  loadThreads,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  muteNotifications,
  selectThread,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeThreads,
  removeThreadsLabel
} from './threads';
import { addEmails, loadEmails } from './emails';
import { addLabels, addLabel, loadLabels } from './labels';
import {
  addFeeds,
  loadFeeds,
  removeFeed,
  selectFeed,
  toggleMuteFeed
} from './feeds';
import { setThreads, loadThreadsSuggestions } from './threadsSuggestions'

export {
  addEmails,
  addFeeds,
  addLabel,
  addLabels,
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  addUser,
  addUsers,
  filterThreadsByUnread,
  loadEmails,
  loadFeeds,
  loadLabels,
  loadThreads,
  loadThreadsSuggestions,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  muteNotifications,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeFeed,
  removeThreads,
  removeThreadsLabel,
  selectThread,
  selectFeed,
  setThreads,
  toggleMuteFeed
};
