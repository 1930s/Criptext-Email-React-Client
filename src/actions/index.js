import { addUsers, addUser } from './users';
import {
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  filterThreadsByUnread,
  loadThreads,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  selectThread,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeThreads,
  removeThreadsLabel
} from './threads';
import { addEmails, loadEmails } from './emails';
import { addLabels, addLabel, loadLabels } from './labels';
import { addFeeds, loadFeeds, selectFeed } from './feeds';

export {
  addEmails,
  addFeeds,
  addLabel,
  addLabels,
  addThreads,
  addThreadLabel,
  addThreadsLabel,
  addUser,
  addUsers,
  filterThreadsByUnread,
  loadEmails,
  loadFeeds,
  loadLabels,
  loadThreads,
  markThreadsRead,
  moveThreads,
  multiSelectThread,
  removeThread,
  removeThreadLabel,
  deselectThreads,
  searchThreads,
  selectThreads,
  removeThreads,
  removeThreadsLabel,
  selectThread,
  selectFeed
};
