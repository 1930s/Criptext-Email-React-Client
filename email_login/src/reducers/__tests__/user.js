/* eslint-env node, jest */

import userReducer from './../users';
import { addUsers } from '../../actions/index';
import { fromJS } from 'immutable';

jest.mock('../../utils/electronInterface');

describe('User actions:', () => {
  it('Add a new user to state', () => {
    const prevState = fromJS({
      '0': {
        id: '0',
        name: 'Previous User',
        nickname: 'prevuser',
        email: 'prevuser@criptext.com'
      }
    });
    const newUser = {
      '1': {
        id: '1',
        name: 'Test User',
        nickname: 'testuser',
        email: 'test1@criptext.com'
      }
    };
    const action = addUsers(newUser);
    const state = userReducer(prevState, action);
    expect(state).toMatchSnapshot();
  });
});
