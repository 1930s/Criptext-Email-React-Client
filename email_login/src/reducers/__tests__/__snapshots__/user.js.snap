// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User actions: Add a new user to state 1`] = `
Immutable.Map {
  0: Immutable.Map {
    id: "0",
    name: "Previous User",
    nickname: "prevuser",
    email: "prevuser@criptext.com",
  },
  1: Immutable.Map {
    id: "1",
    name: "Test User",
    nickname: "testuser",
    email: "test1@criptext.com",
  },
}
`;
